instrumentation/spring/spring-integration-4.1/testing/src/main/java/AbstractRabbitProducerConsumerTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static io.opentelemetry.instrumentation.testing.GlobalTraceUtil.runWithSpan;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.messaging.support.MessageBuilder;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+public abstract class AbstractRabbitProducerConsumerTest {
+
+  static GenericContainer rabbitMqContainer;
+  static ConfigurableApplicationContext producerContext;
+  static ConfigurableApplicationContext consumerContext;
+
+  public void startRabbit(Class<?> additionalContext) {
+    rabbitMqContainer =
```

Comment:
According to https://scans.gradle.com/s/4bpfrmndfo234/console-log/task/:instrumentation:spring:spring-integration-4.1:testing:compileJava?anchor=15&page=1
There are problems in following definition, it can be solved by referring to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/2af49b4c044db526dd0741eccfe0bdc07baeda71/instrumentation/rabbitmq-2.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/AbstractRabbitMqTest.java#L35